src/tag/index.js

```diff
@@ -41,14 +41,15 @@ class Tag extends PureComponent {
 
     const tagId = `${id}_tag`
     const className = cx('tag-remove', { readOnly }, { disabled })
-    const onClick = !readOnly && !disabled ? this.handleClick : undefined
-    const onKeyDown = !readOnly && !disabled ? this.onKeyDown : undefined
-    const onKeyUp = !readOnly && !disabled ? this.onKeyUp : undefined
+    const isDisabled = readOnly || disabled
+    const onClick = !isDisabled ? this.handleClick : undefined
+    const onKeyDown = !isDisabled ? this.onKeyDown : undefined
+    const onKeyUp = !isDisabled ? this.onKeyUp : undefined
 
     return (
       <span className={cx('tag')} id={tagId}>
         {label}
-        <button onClick={onClick} onKeyDown={onKeyDown} onKeyUp={onKeyUp} className={className} type="button" aria-label="Delete" aria-labelledby={tagId} aria-disabled={readOnly || disabled}>
+        <button onClick={onClick} onKeyDown={onKeyDown} onKeyUp={onKeyUp} className={className} type="button" aria-label="Delete" aria-labelledby={tagId} aria-disabled={isDisabled}>
```

Comment:
Do we need both `aria-label` and `aria-labelledby`?

Reply:
When I inspected it we get the "{Tag} Delete" as description.
Otherwise it would be unclear what the button does.

Reply:
> I guess we only could set a aria-label="Delete {Label}" instead?

Sure, let's roll with that. Umm, do think we should say `Delete {Label} tag`?

Reply:
Should be able to set by prop also I guess. But should that be in the format "Delete {Label} tag" in that case?

Reply:
![bild](https://user-images.githubusercontent.com/17863113/55689998-2b676280-598c-11e9-88f0-fa9905ccabd7.png)
From the pill list here, I'll guess we'll keep it then. But perhaps one should be able to change it by prop (localization or other reasons).

Reply:
![bild](https://user-images.githubusercontent.com/17863113/55689998-2b676280-598c-11e9-88f0-fa9905ccabd7.png)
From the pill list [here](https://www.w3.org/TR/2017/NOTE-wai-aria-practices-1.1-20171214/examples/grid/LayoutGrids.html), I'll guess we'll keep it then. But perhaps one should be able to change it by prop (localization or other reasons).

Reply:
Hmmm, I'm not sure why both are needed. Will look up the reasons later. For now, let's keep them both.

Reply:
> But should that be in the format "Delete {Label} tag" in that case?

Think so. I was imagining how the screen reader would read it out and so, adding "tag" would help or not.

Reply:
I suggest we go with the matching example from w3 (which uses Remove + two label references by id) .Added possibility to control the label by prop though.

Reply:
Yeah, let's do that. 

> We could default it to "Remove tag"?

Default which one?

Reply:
`labelRemove = "Remove"`	> "Remove tag"

Reply:
Oh, I see. I missed those new props somehow. 

I don't think it `labelRemove` should be an option. Are you thinking this in terms of i18n? 

And can you tell me more about `label` prop? Why is it needed?

Reply:
For i18n and validation. It's in the readme/typings. When the trigger has a role it also will require a label just as the input.

Reply:
What do you say we tackle i18n separately? And if possible in a way that doesn't require more props. I already think that we're approaching a "prop overload" (I'm wondering if it's time to split the component into separate packages) and so if we go the prop way, I fear we may have to introduce more props later.

Reply:
How about starting to bundle props in objects instead, Mode/enum instad of separate booleans, `texts = { placeholder, label, labelRemove }` etc?

We would need a possibility to set the label since "Remove Blåval" and "Remove Bläckfisk" is not something we want presented for our potential screen reader users. Having to modify the dom because the component does not provide a way feels a bit odd. It's the only "static" text then which you would not be able to change without the prop.

Reply:
> How about starting to bundle props in objects instead, Mode/enum instad of separate booleans, texts = { placeholder, label, labelRemove } etc?

That seems like a good start. Let's go with that for now. We won't be backward compatible but I think that's Ok.

Reply:
I'll try to fix the migration step guide in the onAction pr so we can get that through and fill it with prop-changes then.

Reply:
The bundle of props is now reverted according [the other PR comment](https://github.com/dowjones/react-dropdown-tree-select/pull/217#issuecomment-483472154)